src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -410,8 +413,33 @@ public class GenerateProtoTask extends DefaultTask {
       }
     }
 
-    cmd.addAll protoFiles
-    logger.log(LogLevel.INFO, cmd.toString())
+    int lengthLimit = getCmdLengthLimit()
+
+    String baseCmd = cmd.join(" ") + " "
+
+    StringBuilder currentCmd = new StringBuilder(baseCmd)
+    int currentFileCount = 0
+    for (File proto : protoFiles) {
+      currentCmd.append(proto.toString()).append(" ")
+      currentFileCount++
+      // Offset to account for trailing space
+      if (currentCmd.length() - 1 > lengthLimit) {
```

Comment:
This won't prevent currentCmd from exceeding the limit at the beginning of the next iteration, and `-1` should be `+1`.
The condition should be `currentCmd.length() + 1 + nextProtoFile.toString().length() > lengthLimit`